pkg/agent/openflow/client.go

```diff
@@ -170,61 +177,100 @@ func (c *client) UninstallPodFlows(containerID string) error {
 }
 
 func (c *client) InstallClusterServiceCIDRFlows(serviceNet *net.IPNet, gatewayOFPort uint32) error {
-	return c.flowOperations.Add(c.serviceCIDRDNATFlow(serviceNet, gatewayOFPort))
+	return c.flowOperations.Add(c.serviceCIDRDNATFlow(serviceNet, gatewayOFPort, cookie.Service))
 }
 
 func (c *client) InstallGatewayFlows(gatewayAddr net.IP, gatewayMAC net.HardwareAddr, gatewayOFPort uint32) error {
-	if err := c.flowOperations.Add(c.gatewayClassifierFlow(gatewayOFPort)); err != nil {
+	if err := c.flowOperations.Add(c.gatewayClassifierFlow(gatewayOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.gatewayIPSpoofGuardFlow(gatewayOFPort)); err != nil {
+	} else if err := c.flowOperations.Add(c.gatewayIPSpoofGuardFlow(gatewayOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.gatewayARPSpoofGuardFlow(gatewayOFPort)); err != nil {
+	} else if err := c.flowOperations.Add(c.gatewayARPSpoofGuardFlow(gatewayOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.l3ToGatewayFlow(gatewayAddr, gatewayMAC)); err != nil {
+	} else if err := c.flowOperations.Add(c.l3ToGatewayFlow(gatewayAddr, gatewayMAC, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(gatewayMAC, gatewayOFPort)); err != nil {
+	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(gatewayMAC, gatewayOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.localProbeFlow(gatewayAddr)); err != nil {
+	} else if err := c.flowOperations.Add(c.localProbeFlow(gatewayAddr, cookie.Default)); err != nil {
 		return err
 	}
 	return nil
 }
 
 func (c *client) InstallTunnelFlows(tunnelOFPort uint32) error {
-	if err := c.flowOperations.Add(c.tunnelClassifierFlow(tunnelOFPort)); err != nil {
+	if err := c.flowOperations.Add(c.tunnelClassifierFlow(tunnelOFPort, cookie.Default)); err != nil {
 		return err
-	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(globalVirtualMAC, tunnelOFPort)); err != nil {
+	} else if err := c.flowOperations.Add(c.l2ForwardCalcFlow(globalVirtualMAC, tunnelOFPort, 0)); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (c *client) Initialize() error {
+func (c *client) Initialize(brClient ovsconfig.OVSBridgeClient) error {
```

Comment:
I think it should be better to make brClient as a property of Openflow client. When Openflow switch is disconnected or reconnected, Openflow client might need to to renew the round ID. So it might be accessed outside this function. What do you think @antoninbas ?

Reply:
I think the roundID should live along with the agent but not a connection, since we don't need to clean flow entries if we just have a reconnection, right?

Reply:
We can't predict if the Openflow entries have been changed after a reconnection. Resync seems to be a secure solution after a reconnection. And it also helps to maintain a single code between  agent start and reconnection.

Reply:
Good idea, moved.